### ISSUE TITLE
Issue with PKG_CONFIG_PATH resolving past 2.0.0

### ISSUE THREAD

**Issue by 1480c1** [pkgconf/pkgconf#326]
Steps to reproduce:
1. Prep two hypothetical prefixes (could be a local install and the normal `/usr/lib/pkgconfig` folder)
```bash
for folder in a b; do
    mkdir -p $folder
    sed 's/PREFIX/'$folder'/' > $folder/c.pc << 'EOF'
prefix=/PREFIX
exec_prefix=${prefix}
includedir=${prefix}/include
libdir=${exec_prefix}/lib

Name: c
Description: c
Version: 1.0.0
Cflags: -I${includedir}
Libs: -L${libdir} -lc
Libs.private: -ld
EOF
done
export PKG_CONFIG_PATH="$PWD/a:$PWD/b"
```

2. run pkgconf < 2.0.0
```bash
$ pkgconf --static --libs c
-L/a/lib -lc -ld
```
3. run pkgconf >= 2.0.0
```bash
$ ./bui/pkgconf --static --libs c
-L/b/lib -lc -ld
```

Offending commit after bisecting seems to be https://github.com/pkgconf/pkgconf/commit/384ade5f316367f15da9dbf09853e06fe29bf2bf and reverting that restores pre-2.0.0 behavior, however, I am not entirely sure the motivation behind that commit, so I can't tell if reverting will break something else.

I can't submit to <https://gitea.treehouse.systems/ariadne/pkgconf> right now, as it's giving me error 500.

**Comment by lazka** [comment]
Since the commit message mentions `--with-path` and the docs there state "Adds a new module search path to pkgconf's dependency resolver. Paths added in this way are given preference before other paths." I'd guess that the changed behaviour for other cases was an oversight.

**Comment by dg0yt** [comment]
I guess this issue breaks the pkgconf 2.0 update attempts in vcpkg (https://github.com/microsoft/vcpkg/pull/33055, https://github.com/microsoft/vcpkg/pull/33729, https://github.com/microsoft/vcpkg/pull/34175): 
Similar to `PKG_CONFIG_PATH="$PWD/a:$PWD/b"` above, I observe unexpected  debug variant pc file results for "release:debug" style pkgconfig path order.

**Comment by dg0yt** [comment]
IIUC all list processing in `pkgconf_path_split` (`pkgconf_path_build_from_environ`) now results in reversed lists, doesn't it?

**Comment by dg0yt** [comment]
IIUC `--with-path` is applied early (before reading PKG_CONFIG_PATH) so it wouldn't need to insert. Insertion at begin is the wrong operation also when the option is given multiple times.
So just revert the change, and add a test?
https://github.com/pkgconf/pkgconf/compare/master...dg0yt:pkgconf:list-order?expand=1

**Comment by kaniini** [comment]
Fixed in 652aff97906d7eadca6fe727b16f6511d8dbfe05
